### PR TITLE
Hot fire test states

### PR DESCRIPTION
**Pull request by ZachLegros**

### CODE REVIEW

**Comment by ajayyy** [comment]
Depends on https://github.com/uorocketry/rocket-code-2020/pull/25

File: projects/OctoberSky1/config/UOStateMachine.cpp [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+	duration_ns timeSinceEntry = std::chrono::steady_clock::now() - entryTime;
+	if (timeSinceEntry >= duration_ns(targetTime))
+	{
+		InternalEvent(state);
+
+		return true;
+	}
+
+	return false;
+}
```
Comment (ajayyy) [inline projects/OctoberSky1/config/UOStateMachine.cpp:271]: Is there a reason you aren't using the function from StateMachine.cpp?

Comment (ZachLegros) [reply projects/OctoberSky1/config/UOStateMachine.cpp:271]: Me and Seb didn't know that the functions were transferred in State Machine.cpp so we copied these implementations from past commits. I'll remove them and use the ones in StateMachine.cpp.

**Review by sebastiengrd — APPROVED** [review]

File: sbgECom/src/IO/Logger.h [hunk 2]
```diff
@@ -1,40 +1,41 @@
...
 
 private:
-    //pop queue and log the data from rocketState on logging thread
+	//pop queue and log the data from sensorsData on logging thread
 	void dequeueToFile();
 
-	void writeHeader(std::ofstream& file);
-	void writeData(std::ofstream& file, const rocketState& currentState);
+	void writeHeader(std::ofstream &file);
+	void writeData(std::ofstream &file, const sensorsData &currentState);
```
Comment (ajayyy) [inline sbgECom/src/IO/Logger.h:31]: Reference symbol moved

**Pull request merged**